### DISCUSSION TITLE
How do I rework keymappings? I want to change 60 keymappings.

### DISCUSSION THREAD

**Discussion by Folaht** [LazyVim/LazyVim#4858] [Q&A]
Lunarvim has been discontinued, so I'm moving to here. 
I use an orthogonal keyboard with a Dvorak layout, so I have all the basic keymappings different. 
How do I rework all these keymappings into Lazyvim?  
I've got like 60 of them.

I'm reading about keymappings, but it just confuses me where to even start.
There's an empty `lua/config/keymaps.lua` which I can find.  
There's a `lazyvim.config.keymaps` which I can't find anywhere.  
And there's plugin and LSP keymaps, which I don't believe is relevant for my situation?  

I'm just really confused. It used to be so easy to just tag on this file below..

`my_keymaps.lua`
```
local map = vim.api.nvim_set_keymap
local options = { noremap = true, silent = true }

-- character movement (8)

map('', 'h', 'h', options)            
map('', 'H', '^', options)         
map('', 't', 'k', options)          
map('', 'T', '<PageUp>', options)     
map('', 'n', 'j', options)          
map('', 'N', '<PageDown>', options)  
map('', 's', 'l', options)            
map('', 'S', 'g_l', options)         

-- word movement (8)

map('', 'g', 'g', options)
map('', 'G', 'G', options)
map('', 'c', 'b', options)
...
```

**Comment by yxian9** [comment]
you can just paste `my_keymaps.lua` into  `lua/config/keymaps.lua` lazyvim will load it when start.

**Comment by cerebrl** [comment]
There's more than one way to do this. I can't claim any one way is right, but you can check out my `nvim` config repo here: https://github.com/cerebrl/nvim/tree/main. 

For me, I try to keep keymap config close to what I'm remapping. If I'm changing the keymaps for a specific plugin, I do the changes in the plugin's lua file, like this: https://github.com/cerebrl/nvim/blob/main/lua/plugins/utilities.lua.

If I'm creating my own keymaps, or changing a core nvim keymap, I place that config in the `/config` directory, like this: https://github.com/cerebrl/nvim/blob/main/lua/config/keymaps.lua.

Again, I'm no expert, but this works for me.
